### PR TITLE
App template updates

### PR DESCRIPTION
**Pull request by ianedwards**
## What does this PR do?

- The app template form was being populated with values from the latest revision regardless of whether or not these values were actually present in the app template
- This was causing some confusion since preview envs weren't following what showed up in the app template on the dashboard

### CODE REVIEW

File: dashboard/src/main/home/cluster-dashboard/preview-environments/v2/setup-app/SetupApp.tsx [hunk 1]
```diff
@@ -42,7 +86,13 @@ const SetupApp: React.FC<Props> = ({ location }) => {
               disableLineBreak
             />
             <DarkMatter />
-            <AppTemplateForm />
+            {match(templateRes)
+              .with({ status: "loading" }, () => <Loading />)
+              .with({ status: "success" }, ({ data }) => {
+                console.log("data", data);
```
Comment (ferozemohideen) [inline dashboard/src/main/home/cluster-dashboard/preview-environments/v2/setup-app/SetupApp.tsx:92]: remove log

File: dashboard/src/main/home/cluster-dashboard/preview-environments/v2/setup-app/SetupApp.tsx [hunk 2]
```diff
@@ -24,6 +32,42 @@ const SetupApp: React.FC<Props> = ({ location }) => {
 
   const appName = params.appName;
 
+  const templateRes = useQuery(
+    ["getAppTemplate", currentProject?.id, currentCluster?.id, appName],
+    async () => {
+      if (!currentProject || !currentCluster || !appName) {
```
Comment (ferozemohideen) [inline dashboard/src/main/home/cluster-dashboard/preview-environments/v2/setup-app/SetupApp.tsx:38]: might want to check currentCluster !== -1

**Review by ferozemohideen — APPROVED** [review]

File: dashboard/src/main/home/cluster-dashboard/preview-environments/v2/setup-app/SetupApp.tsx [hunk 3]
```diff
@@ -24,6 +32,42 @@ const SetupApp: React.FC<Props> = ({ location }) => {
...
+
+        return PorterApp.fromJsonString(atob(data.template_b64_app_proto), {
+          ignoreUnknownFields: true,
+        });
+      } catch (err) {
+        return null;
+      }
+    },
+    {
+      enabled: !!currentProject && !!currentCluster && !!appName,
```
Comment (ferozemohideen) [inline dashboard/src/main/home/cluster-dashboard/preview-environments/v2/setup-app/SetupApp.tsx:67]: is this gonna refetch on window focus (and potentially change status state)

Comment (ianedwards) [reply dashboard/src/main/home/cluster-dashboard/preview-environments/v2/setup-app/SetupApp.tsx:67]: will change

**Pull request merged**